maps/src/test/java/org/odk/collect/maps/layers/OfflineMapLayersPickerTest.kt

```diff
@@ -793,6 +798,35 @@ class OfflineMapLayersPickerTest {
         onView(withId(R.id.layers)).check(matches(isDisplayed()))
     }
 
+    @Test
+    fun `the confirmation dialog is dismissed o activity recreation`() {
+        val scenario = launchFragment()
+
+        uris.add(Uri.parse("blah"))
+        Interactions.clickOn(withText(string.add_layer))
+
+        scenario.onFragment {
+            assertThat(
+                it.childFragmentManager.findFragmentByTag(OfflineMapLayersImporter::class.java.name),
+                instanceOf(OfflineMapLayersImporter::class.java)
+            )
+        }
+
+        scenario.recreate()
+
+        scenario.onFragment {
+            assertThat(
+                it.childFragmentManager.findFragmentByTag(OfflineMapLayersImporter::class.java.name),
+                equalTo(null)
+            )
+        }
+    }
+
+    private fun toggleLayer(position: Int) {
+        onView(withRecyclerView(R.id.layers).atPositionOnView(position, R.id.arrow)).perform(click())
+        WaitFor.wait250ms()
```

Comment:
Could we not wait for something here (`WaitFor.waitFor` for example) instead of just waiting a fixed amount of time? That doesn't give us any guarantee that we've waited enough. 

Also, did you try things like using the `InstantTaskExecutor` or maybe even checking if animations are somehow still enabled?

Reply:
> Could we not wait for something here (WaitFor.waitFor for example) instead of just waiting a fixed amount of time? That doesn't give us any guarantee that we've waited enough.

Yeah, we can use it.

> Also, did you try things like using the InstantTaskExecutor or maybe even checking if animations are somehow still enabled?

I did, but nothing worked.